geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/GMSMembership.java

```diff
@@ -677,10 +658,12 @@ private void removeWithViewLock(ID dm, boolean crashed, String reason) {
       return; // Explicit deletion, no upcall.
     }
 
-    if (!shutdownMembers.containsKey(dm)) {
-      // if we've received a shutdown message then DistributionManager will already have
-      // notified listeners
-      listener.memberDeparted(dm, crashed, reason);
+    synchronized (shutdownMembers) {
+      if (!shutdownMembers.containsKey(dm)) {
+        // if we've received a shutdown message then DistributionManager will already have
+        // notified listeners
+        listener.memberDeparted(dm, crashed, reason);
```

Comment:
Is it necessary to hold the lock on shutdownMembers while invoking a listener?  The point of this check is to avoid invoking memberDeparted if a shutdown message from **dm** has already been processed.  In that case we've already invoked memberDeparted on the listener.